### ISSUE TITLE
[Bug]: [OneSignalAttachmentHandler trimURLSpacing:] crash

### ISSUE THREAD

**Issue by DenTelezhkin** [OneSignal/OneSignal-iOS-SDK#1400]
### What happened?

We are getting a lot of crash reports from Apple (thousands) about this issue. This crash is in notification extension, so not necessarily visible to users, but may cause push notifications to malfunction (particularly in-app messages).

### Steps to reproduce?

```Markdown
Unfortunately, I don't have any, I was not able to reproduce this issue on my side.
```

### What did you expect to happen?

Notification extension to not crash. Unfortunately, it does, attaching crash log:

[2024-04-08_11-15-02.7507_+0000-107575d31fde83b8356216cb6975692313f1e198.crash.zip](https://github.com/OneSignal/OneSignal-iOS-SDK/files/14906880/2024-04-08_11-15-02.7507_%2B0000-107575d31fde83b8356216cb6975692313f1e198.crash.zip)

### OneSignal iOS SDK version

5.0.5

### iOS version

```Markdown
* iOS 17, however this may be because most of app clients are on iOS 17.
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by tom-willmowski** [comment]
I've got the same issue. Can you help?

**Comment by emawby** [comment]
Fix is in #1411
